packages/web-react/src/components/Grid/GridItem.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { CSSProperties, ElementType } from 'react';
+import classNames from 'classnames';
+import { useStyleProps } from '../../hooks';
+import { GridColumn, GridColumnBreakpoints, SpiritGridItemProps } from '../../types';
+import { useGridItemStyleProps } from './useGridItemStyleProps';
+
+interface GridItemCSSProperties extends CSSProperties {
+  [key: string]: string | undefined | number;
+}
+
+const setStyleProperty = (styleObject: GridItemCSSProperties, propertyName: string, value: string | undefined) => {
+  (styleObject as Record<string, string | undefined>)[propertyName] = value;
+};
+
+const GridItem = <T extends ElementType = 'div'>(props: SpiritGridItemProps<T>): JSX.Element => {
+  const { elementType: ElementTag = 'div', children, columnEnd, columnSpan, columnStart, ...restProps } = props;
+  const { classProps, props: modifiedProps } = useGridItemStyleProps(restProps);
+  const { styleProps, props: otherProps } = useStyleProps(modifiedProps);
+
+  const gridItemStyle: GridItemCSSProperties = {};
+
+  const setGridItemStyle = (propValue: GridColumn, baseVarName: string) => {
+    if (typeof propValue === 'number') {
+      const propName = `--${baseVarName}`;
+      setStyleProperty(gridItemStyle, propName, propValue.toString());
+    } else if (typeof propValue === 'object' && propValue !== null) {
+      Object.keys(propValue).forEach((key) => {
+        const suffix = key === 'mobile' ? '' : `-${key}`;
+        const propName = `--${baseVarName}${suffix}`;
+        setStyleProperty(gridItemStyle, propName, propValue[key as keyof GridColumnBreakpoints]?.toString());
+      });
+    }
+  };
+
+  if (columnEnd) {
+    setGridItemStyle(columnEnd, 'grid-item-column-end');
```

Comment:
I would rather change the name and the value to match the key:value order, e.g. `setGridItemStyle('grid-item-column-end', columnEnd);`. I think it is much readable to know what I am setting first.